azure-devops-extension/run-automator-build-task/src/sdk/tools-api/client.ts

```diff
@@ -0,0 +1,94 @@
+import axios, { AxiosResponse } from "axios";
+
+import { MAX_TIME_SPEND_ON_POLLING, POLLING_INTERVAL } from "@common/constants";
+import { sleep } from "@common/helpers/sleep.helper";
+import { BuildNotFinishedError } from "@sdk/tools-api/errors/build-not-finished.error";
+import { BuildResultPollingTimeoutError } from "@sdk/tools-api/errors/build-result-polling-timeout.error";
+import { IPollBuildResultsResponse } from "@sdk/tools-api/interfaces/poll-build-results-response.interface";
+import { IStartBuildResponse } from "@sdk/tools-api/interfaces/start-build-response.interface";
+
+enum ToolsAPIRoute {
+  StartBuild,
+  PollBuildResults,
+}
+
+interface IRequestParameters {
+  route: ToolsAPIRoute;
+  body: Record<string, unknown>;
+}
+
+export interface IToolsAPIClientOptions {
+  baseURL: string;
+  startBuildPath: string;
+  pollBuildResultsPath: string;
+}
+
+export class ToolsAPIClient {
+  private routes: Map<ToolsAPIRoute, string>;
+
+  constructor(options: IToolsAPIClientOptions) {
+    this.routes = this.initRoutes(options);
+  }
+
+  private initRoutes({
+    baseURL,
+    startBuildPath,
+    pollBuildResultsPath,
+  }: Pick<IToolsAPIClientOptions, "baseURL" | "startBuildPath" | "pollBuildResultsPath">): Map<ToolsAPIRoute, string> {
+    const routes = new Map<ToolsAPIRoute, string>();
+    routes.set(ToolsAPIRoute.StartBuild, `${baseURL}${startBuildPath}`);
+    routes.set(ToolsAPIRoute.PollBuildResults, `${baseURL}${pollBuildResultsPath}`);
+    return routes;
+  }
+
+  private async makePostRequest<T>(parameters: IRequestParameters): Promise<AxiosResponse<T>> {
+    const routeURL = this.routes.get(parameters.route);
+    if (!routeURL) throw new Error("Route URL not available");
+    return axios.post<T>(routeURL, JSON.stringify(parameters.body), {
+      headers: {
+        "Content-Type": "application/json",
+      },
+    });
+  }
+
+  public async startBuild(authToken: string, testSuiteId: string, ciBuildId?: string): Promise<string> {
+    const triggerResponse = await this.makePostRequest<IStartBuildResponse>({
+      route: ToolsAPIRoute.StartBuild,
+      body: {
+        authToken,
+        testSuiteId,
+        ciBuildId,
+      },
+    });
+    return triggerResponse.data.buildId;
+  }
+
+  public async poll(token: string, buildId: string, currentRunTime = 0): Promise<void> {
+    try {
+      const didTestsPass = await this.getResults(token, buildId);
+      console.log(`DeepCrawl Tests ${didTestsPass ? "Passed" : "Failed"}`);
```

Comment:
nit: I would rather not use console anywhere. It's part of `dom` which isn't present in tsconfig `lib` value at the moment. It's nicer to use dedicated logger like `pino` or any other that supports standard output if the logging to console is needed.

Reply:
done, nice catch :)